trackerquery_test.go

```diff
@@ -119,3 +119,67 @@ func TestUpdateTrackerQuery(t *testing.T) {
 		t.Errorf("Type has changed has from %s to %s", tqresult.Schedule, updated.Schedule)
 	}
 }
+
+// This test ensures that List does not return NIL items.
+func TestTrackerQueryListItemsNotNil(t *testing.T) {
+	resource.Require(t, resource.Database)
+	ts := models.NewGormTransactionSupport(DB)
+	repo := remoteworkitem.NewTrackerRepository(ts)
+	controller := TrackerController{ts: ts, tRepository: repo, scheduler: rwiScheduler}
+	payload := app.CreateTrackerAlternatePayload{
+		URL:  "http://api.github.com",
+		Type: "github",
+	}
+	_, result := test.CreateTrackerCreated(t, nil, nil, &controller, &payload)
+	t.Log(result.ID)
+	tqts := models.NewGormTransactionSupport(DB)
+	tqrepo := remoteworkitem.NewTrackerQueryRepository(tqts)
+	tqController := TrackerqueryController{ts: tqts, tqRepository: tqrepo, scheduler: rwiScheduler}
+	tqpayload := app.CreateTrackerQueryAlternatePayload{
+
+		Query:     "is:open is:issue user:arquillian author:aslakknutsen",
+		Schedule:  "15 * * * * *",
+		TrackerID: result.ID,
+	}
+	_, item1 := test.CreateTrackerqueryCreated(t, nil, nil, &tqController, &tqpayload)
+	_, item2 := test.CreateTrackerqueryCreated(t, nil, nil, &tqController, &tqpayload)
+
+	_, list := test.ListTrackerqueryOK(t, nil, nil, &tqController, nil, nil)
+	for _, tq := range list {
+		if tq == nil {
+			t.Error("Returned Tracker Query found nil")
+		}
+	}
+	test.DeleteTrackerqueryOK(t, nil, nil, &tqController, item1.ID)
+	test.DeleteTrackerqueryOK(t, nil, nil, &tqController, item2.ID)
+}
+
+// This test ensures that ID returned by Show is valid.
+// refer : https://github.com/almighty/almighty-core/issues/189
+func TestCreateTrackerQueryValidId(t *testing.T) {
```

Comment:
@DhritiShikhar I'll assume what you're trying to say is that 'TestShow' is already calling the 'Delete' endpoint. But.. 1; Delete is called but non of it's behavior is verified. It might as well be an empty method. 2. You test functions in isolation to verify they behave as intended. When TestX fail, but TestY doesn't, you can assume Y is not the cause of X 's failure (assuming X use Y).